.github/workflows/release.yaml

```diff
@@ -3,59 +3,55 @@ name: Release
 on:
   push:
     tags:
-      - "*"
+      - v[0-9]+.*
 jobs:
-  build:
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest]
-        rust: [stable]
+  body:
+    runs-on: ubuntu-latest
     permissions:
       contents: write
-
-    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
-        uses: actions/checkout@master
-
-      - name: Install just
-        uses: taiki-e/install-action@just
-
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-
-      - name: Setup build env for Linux
-        if: matrix.os == 'ubuntu-latest'
-        run: |
-          sudo apt update
-          sudo apt install -y musl-tools gcc
-          rustup target add x86_64-unknown-linux-musl
-
-      - name: Build for Linux
-        if: matrix.os == 'ubuntu-latest'
-        run: |
-          just build-release-linux
-          cp target/x86_64-unknown-linux-musl/release/tenere tenere-x86_64-linux-musl
-
-      - name: Build for macos
-        if: matrix.os == 'macos-latest'
-        run: |
-          rustup target add aarch64-apple-darwin
-          just build-release-macos
-          cp target/x86_64-apple-darwin/release/tenere tenere-x86_64-macos
-          cp target/aarch64-apple-darwin/release/tenere tenere-aarch64-macos
-
+        uses: actions/checkout@v4
       - name: Extract release notes
-        if: matrix.os == 'ubuntu-latest'
+        uses: ffurrer2/extract-release-notes@v2
         id: release_notes
-        uses: ffurrer2/extract-release-notes@v1
+      - name: Upload Body
+        uses: softprops/action-gh-release@v2
+        with:
+          body: ${{ steps.release_notes.outputs.release_notes }}  
 
-      - name: Release
-        uses: softprops/action-gh-release@v1
+  build:
+    permissions:
+      contents: write
+    continue-on-error: true
+    strategy:
+      matrix:
+        include:
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+          - target: aarch64-apple-darwin
+            os: macos-latest
+          - target: x86_64-apple-darwin
+            os: macos-latest
+          - target: x86_64-pc-windows-msvc
+            os: windows-latest
+          - target: aarch64-pc-windows-msvc
+            os: windows-latest
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Target
+        run: rustup target add ${{ matrix.target }}
```

Comment:
better to use https://github.com/dtolnay/rust-toolchain instead

Reply:
It is archived. Not maintained. 
Why is calling an archived not maintained GitHub action better that calling 
`rustup` directly?

Reply:
where do you see it is archived ? 
It makes sure you have all the necessary tools to compile

Reply:
You see it on their GitHub.
https://github.com/actions-rs/toolchain

Reply:
Nevermind. I looked at action-rs toolchain.
Though my point stands. 
Could you tell me why you believe it is better to call a third party github action, that calls
`rustup` instead of calling `rustup` yourself?

Reply:
in this file yes you are right we can use `rustup` directly. but if you look at `ci.yaml` I used it to install additional tools. 
So, the only argument I have for using it here is just to be consistent with the other workflows.

Reply:
Well. I won't touch `ci.yaml`. I believe it is always better to use the first party tool instead. If you really really wish, then I'll change it, tho it is really fussy.

Reply:
I don't have strong opinion here. you can keep `rustup` if you prefer :)

Reply:
It actually looks so much worse. :(